knn-banking-product-recommender/deployment/README.md

```diff
@@ -66,3 +66,48 @@ This step builds the Kserve predictor image that contains your model.
     $responseObject = $response.Content | ConvertFrom-Json
     $responseObject | ConvertTo-Json -Depth 10
     ```
+
+## Gradio App
+
+The Gradio app is a simple graphical interface for testing the model. It is not required for deployment, but it can be added to the Highwind Use Case to allow for easier testing.
+
+### Local testing
+
+> All commands below are run from this directory.
+> You can also test how the app works without running inference on the deployed Highwind Use Case by setting function of the function of the `submit_button.click` to `get_mock_product_recommendation` instead of `get_product_recommendation` (set this in `demo.py`).
+
+1. Install the dependencies in a virtual environment
+
+    > This section requires Python 3.12 or higher
+
+    ```bash
+    python -m venv .gradio.venv && source .gradio.venv/bin/activate
+    pip install -r gradio.requirements.txt
+    ```
+
+1. Uncomment the last line of the `demo.py` file so that the app launches locally when following the next step.
+
+    > ⚠️ Warning: Do not forget to comment the `demo.launch()` line of the `demo.py` before deploying to Highwind. This step is only for local testing!
+
+1. Run the app (from the virtual environment)
+
+    ```bash
+    python demo.py
```

Comment:
I think something to note here is that without specifying any environment variables for the Highwind SDK, it will automatically attempt to communicate with UAT/Zindi. Perhaps it is worth mentioning that?